apps/settings/js/modules/addon_manager.js

```diff
@@ -0,0 +1,403 @@
+        apps.forEach((app) => {
+          var manifest = this._getManifest(app);
+
+          // Ignore apps that are themselves add-ons
+          // XXX: Will themes have a role, and should we ignore them too?
+          if (manifest.role === 'addon') {
+            return;
+          }
+
+          // If the addon doesn't have high enough privileges to affect this app
+          // then we can just return now.
+          if (!this._privilegeCheck(addon, app)) {
+            return;
+          }
+
+          // Get the URL of the app origin plus its launch path to
+          // compare against each of the filters.
+          // XXX: Note that we and do not check the paths used by
+          // activity handlers or any other paths in the manifest.
+          var launchPath = manifest.launch_path || '';
+          var launchURL = new URL(launchPath, app.origin).href;
+          var appliedCustomizations = [];
+
+          // Now loop through the filters to see what customizations are
+          // applied to this app
+          for(var i = 0; i < filters.length; i++) {
+            var filter = filters[i];
+            if (filter.test(launchURL)) {
+              appliedCustomizations.push(customizations[i]);
+              break;
+            }
+          }
+
+          // If any customizations were applied to this app, then add it
+          // to the list of customizedApps
+          if (appliedCustomizations.length > 0) {
+            customizedApps.push({
+              app: app,
+              customizations: appliedCustomizations
+            });
+          }
+        });
+
+        return customizedApps;
+      });
+    },
+
+    /**
+     * Check whether an addon is enabled.
+     *
+     * @access public
+     * @memberOf AddonManager.prototype
+     * @param {DOMApplication} addon
+     * @returns {Boolean}
+     */
+    isEnabled: function(addon) {
+      return this._isAddon(addon) && addon.enabled === true;
+    },
+
+    /**
+     * Check whether an addon can be deleted.
+     *
+     * @access public
+     * @memberOf AddonManager.prototype
+     * @param {DOMApplication} addon
+     * @returns {Boolean}
+     */
+    canDelete: function(addon) {
+      return this._isAddon(addon) && addon.removable === true;
+    },
+
+    /**
+     * Enable an addon
+     *
+     * @access public
+     * @memberOf AddonManager.prototype
+     * @param {DOMApplication} addon
+     */
+    enableAddon: function(addon) {
+      if (this._isAddon(addon)) {
+        mozApps.mgmt.setEnabled(addon, true);
+      } else {
+        return Promise.reject('not an addon');
+      }
+    },
+
+    /**
+     * Disable an addon. Returns a string indicating whether reboot or restart
+     * required for disabling the addon completely.
+     *
+     * @access public
+     * @memberOf AddonManager.prototype
+     * @param {DOMApplication} addon
+     * @returns {Promise String}
+     */
+    disableAddon: function(addon) {
+      if (this._isAddon(addon)) {
```

Comment:
If we are disabling an addon that is currently enabling, is there any problem or error that we should handle here ?